x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsStatsAction.java

```diff
@@ -9,37 +9,31 @@
 import org.elasticsearch.client.node.NodeClient;
 import org.elasticsearch.cluster.metadata.Metadata;
 import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.compatibility.RestApiCompatibleVersion;
 import org.elasticsearch.rest.BaseRestHandler;
 import org.elasticsearch.rest.RestRequest;
 import org.elasticsearch.rest.action.RestToXContentListener;
 import org.elasticsearch.xpack.core.action.util.PageParams;
 import org.elasticsearch.xpack.core.ml.action.GetTrainedModelsStatsAction;
-import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
-import org.elasticsearch.xpack.ml.MachineLearning;
 
 import java.io.IOException;
-import java.util.Collections;
 import java.util.List;
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 import static org.elasticsearch.xpack.core.ml.action.GetTrainedModelsAction.Request.ALLOW_NO_MATCH;
+import static org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig.MODEL_ID;
+import static org.elasticsearch.xpack.ml.MachineLearning.BASE_PATH;
 
 public class RestGetTrainedModelsStatsAction extends BaseRestHandler {
 
     @Override
     public List<Route> routes() {
-        return Collections.emptyList();
-    }
-
-    @Override
-    public List<ReplacedRoute> replacedRoutes() {
         return List.of(
-            new ReplacedRoute(
-                GET, MachineLearning.BASE_PATH + "trained_models/{" + TrainedModelConfig.MODEL_ID.getPreferredName() + "}/_stats",
-                GET, MachineLearning.BASE_PATH + "inference/{" + TrainedModelConfig.MODEL_ID.getPreferredName() + "}/_stats"),
-            new ReplacedRoute(
-                GET, MachineLearning.BASE_PATH + "trained_models/_stats",
-                GET, MachineLearning.BASE_PATH + "inference/_stats"));
+            Route.builder(GET, BASE_PATH + "trained_models/{" + MODEL_ID.getPreferredName() + "}/_stats")
+                .replaces(GET, BASE_PATH + "inference/{" + MODEL_ID.getPreferredName() + "}/_stats", RestApiCompatibleVersion.V_8).build(),
```

Comment:
What is the backport for this going to look like ?

i.e. planning to back port the RestApiCompatibleVersion class, or adjust the replaces method to only take 1 argument ?

Reply:
[n.b. we've talked about this a bit offline]

We've renamed RestApiCompatibleVersion to RestApiVersion, and it is my intention to backport that class to 7.x, and continue to have the arities of `deprecated` and `replaces` that require it, but not to backport any of the actual rest compatible api backing machinery.

That is, the backport for this stanza is going to look more or less exactly the same as this stanza itself already looks.